### PR TITLE
feat(manifest): complete the manifest of all published libraries

### LINKED ISSUE #2940: Add description for each package
Open an issue to properly describe each package

_Originally posted by @breathx in https://github.com/gear-tech/gear/pull/2924#discussion_r1261380341_

### PR DESCRIPTION
**Pull request by clearloop**
Resolves #2940
part of https://github.com/gear-tech/gear/issues/2923
split from #2924 

complete the manifest of for all published libraries

| status | package | categories |  keywords |
|-|-|-|-|
| ✅ | common | `wasm` | `gear`, `wasm` |
| ✅ | common-codegen | `wasm` | `gear`, `wasm`, `codegen` |
| ✅ | core-backend-codegen | `wasm` | `gear`, `wasm`, `codegen` |
| ✅ | core-backend-common | `wasm` | `gear`, `wasm` |
| ✅ | core-backend-wasmi | `wasmi` | `gear`, `wasm`, `wasm-executor` |
| ✅ | core-errors | | `gear` |
| ✅ | core-processor | `wasm` | `gear`, `wasm` |
| ✅ | core | `wasm` | `gear`, `wasm` |
| ✅ | gcli | `command-line-interface`, `command-line-utilities`, `wasm` |  `gear`, `cli`, `wasm` |
| ✅ | gclient | `api-bindings`, `wasm` | `gear`, `sdk`, `api` |
| ✅ | gmeta | `wasm` | `gear`, `wasm` |
| ✅ | gmeta-codegen | `wasm` | `gear`, `codegen` |
| ✅ | gsdk | `wasm`, `api-bindings` | `gear`, `sdk`, `api` |
| ✅ | gsdk-codegen | `api-bindings` | `gear`, `sdk`, `codegen` |
| ✅ | gsys | `wasm`, `api-bindings` | `gear`, `wasm` |
| ✅ | wasm-instrument | `wasm` | `gear`, `wasm` |

@gear-tech/dev

### CODE REVIEW

**Comment by StackOverflowExcept1on** [comment]
we can also fill the `categories` field in `Cargo.toml`
https://doc.rust-lang.org/cargo/reference/manifest.html#the-categories-field

File: Cargo.toml [hunk 1]
```diff
@@ -3,6 +3,8 @@ version = "0.2.2"
 authors = ["Gear Technologies"]
 edition = "2021"
 license = "GPL-3.0"
+homepage = "https://github.com/gear-tech/gear"
```
Comment (shamilsan) [inline Cargo.toml:6]: The homepage could be gear-tech.io, I guess.
```suggestion
homepage = "https://www.gear-tech.io"
```

File: utils/wasm-instrument/Cargo.toml [hunk 2]
```diff
@@ -1,9 +1,14 @@
 [package]
 name = "gear-wasm-instrument"
 version = "0.1.0"
+description = "WASM instrument library for gear programs"
```
Comment (shamilsan) [inline utils/wasm-instrument/Cargo.toml:4]: I prefer `Wasm` as they write at https://webassembly.org

> WebAssembly (abbreviated Wasm) is a binary instruction format for a stack-based virtual machine. 
```suggestion
description = "Wasm instrument library for Gear programs"
```

File: gsdk/codegen/Cargo.toml [hunk 3]
```diff
@@ -1,9 +1,12 @@
 [package]
 name = "gsdk-codegen"
 version = "0.1.0"
+description = "Code generation library for the rust sdk of gear network"
```
Comment (shamilsan) [inline gsdk/codegen/Cargo.toml:4]:
```suggestion
description = "Code generation library for the Rust SDK of Gear network"
```

File: gsdk/Cargo.toml [hunk 4]
```diff
@@ -1,12 +1,14 @@
 [package]
 name = "gsdk"
 version = "0.1.0"
-edition.workspace = true
+keywords = ["gear", "api", "sdk"]
+categories = ["wasm", "api-bindings"]
+description = "Rust SDK of the gear network"
```
Comment (shamilsan) [inline gsdk/Cargo.toml:6]:
```suggestion
description = "Rust SDK of the Gear network"
```

File: gclient/Cargo.toml [hunk 5]
```diff
@@ -1,9 +1,12 @@
 [package]
 name = "gclient"
 version = "0.1.0"
+description = "API client for gear node in rust."
```
Comment (shamilsan) [inline gclient/Cargo.toml:4]:
```suggestion
description = "API client for Gear node in Rust"
```

File: gcli/Cargo.toml [hunk 6]
```diff
@@ -1,15 +1,15 @@
 [package]
 name = "gcli"
+description = "Gear program cli"
```
Comment (shamilsan) [inline gcli/Cargo.toml:3]:
```suggestion
description = "Gear program CLI"
```

File: core-processor/Cargo.toml [hunk 7]
```diff
@@ -1,9 +1,14 @@
 [package]
 name = "gear-core-processor"
 version = "0.1.0"
+description = "The core processor for gear programs"
```
Comment (shamilsan) [inline core-processor/Cargo.toml:4]:
```suggestion
description = "The core processor for Gear programs"
```

File: core-errors/Cargo.toml [hunk 8]
```diff
@@ -1,9 +1,12 @@
 [package]
 name = "gear-core-errors"
 version = "0.1.0"
+description = "Shared errors for gear core libraries"
```
Comment (shamilsan) [inline core-errors/Cargo.toml:4]:
```suggestion
description = "Shared errors for Gear core libraries"
```

File: common/codegen/Cargo.toml [hunk 9]
```diff
@@ -1,9 +1,14 @@
 [package]
 name = "gear-common-codegen"
 version = "0.1.0"
+description = "Code generation library for the gear common library"
```
Comment (shamilsan) [inline common/codegen/Cargo.toml:4]:
```suggestion
description = "Code generation library for the Gear common library"
```

File: gsys/Cargo.toml [hunk 10]
```diff
@@ -1,6 +1,11 @@
 [package]
 name = "gsys"
-version.workspace = true
+version = "0.1.0"
+description = "System calls of gear program"
```
Comment (shamilsan) [inline gsys/Cargo.toml:4]:
```suggestion
description = "System calls of Gear program"
```

File: gmeta/codegen/Cargo.toml [hunk 11]
```diff
@@ -1,7 +1,14 @@
 [package]
 name = "gmeta-codegen"
 version = "0.1.0"
+description = "Code generation library for the metadata of gear program"
```
Comment (shamilsan) [inline gmeta/codegen/Cargo.toml:4]:
```suggestion
description = "Code generation library for the metadata of Gear program"
```

File: gmeta/Cargo.toml [hunk 12]
```diff
@@ -1,7 +1,14 @@
 [package]
 name = "gmeta"
-version.workspace = true
+version = "0.1.0"
+description = "Metadata library for gear programs"
```
Comment (shamilsan) [inline gmeta/Cargo.toml:4]:
```suggestion
description = "Metadata library for Gear programs"
```

**Review by shamilsan — APPROVED** [review]
I am not sure about the versions. It may be simpler to maintain when the version is the only one (in the workspace).

**Comment by clearloop** [comment]
> I am not sure about the versions. It may be simpler to maintain when the version is the only one (in the workspace).

agree, I was considering that I may publish some dirty version in the following PRs, so leaving some spaces for those operations

File: core-processor/Cargo.toml [hunk 13]
```diff
@@ -1,9 +1,14 @@
 [package]
 name = "gear-core-processor"
 version = "0.1.0"
```
Comment (breathx) [inline core-processor/Cargo.toml:3]: version?

**Review by breathx — APPROVED** [review]

**Pull request merged**